### PR TITLE
Build extension on pr

### PR DESCRIPTION
**Pull request by piyalbasu**
This will build a PR preview of `extension` in `releases` and then delete it on PR merge/close

### CODE REVIEW

File: .github/workflows/buildPreview.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: Compile extension PR preview
+
+on: [pull_request]
```
Comment (piyalbasu) [inline .github/workflows/buildPreview.yml:3]: this runs on `create or update pull request`

File: .github/workflows/buildPreview.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+name: Compile extension PR preview
+
+on:
+  pull_request:
+    paths:
+      - "./extension/**"
```
Comment (piyalbasu) [inline .github/workflows/buildPreview.yml:6]: only run this on PR's (and PR updates) that alter `extension/`

File: .github/workflows/buildPreview.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      - name: Build extension
+        uses: actions/setup-node@v1
+        with:
+          node-version: "12"
+      - run: yarn install && yarn build:extension-preview
+      - name: Zip extension build
+        uses: montudor/action-zip@v0.1.0
+        with:
+          args: zip -qq -r ./extension/build/build.zip ./extension/build
+      - name: Delete Release # if it exists
```
Comment (piyalbasu) [inline .github/workflows/buildPreview.yml:24]: delete the old version of the PR preview so we can replace it.

this fails the first time, but this fails gracefully so it does not stop this pipeline

File: .github/workflows/buildPreview.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: pr-release-${{ github.event.number }}
+          release_name: PR Preview ${{ github.event.number }}
+          draft: false
+          prerelease: true
```
Comment (piyalbasu) [inline .github/workflows/buildPreview.yml:40]: store the pr preview as a draft release

**Comment by piyalbasu** [comment]
> Preview build at https://github.com/stellar/lyra/releases/download/pr-release-78/build-pr-preview-78.zip

This PR, although not to `extension` has a PR preview so you can see how this will work in practice. Future PR's to other parts of the repo will not generate a PR preview for the extension

File: .github/workflows/closePreview.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Delete Release
+        uses: dev-drprasad/delete-tag-and-release@v0.1.2
+        with:
+          delete_release: true
+          tag_name: pr-release-${{ github.event.number }}
```
Comment (vcarl) [inline .github/workflows/closePreview.yml:16]: Nice!

**Pull request merged**